### ISSUE TITLE
`nvim_open_win` api, with `relative = "win"` is broken!

### ISSUE THREAD

**Issue by anuvyklack** [neovim/neovim#21752]
### Describe the bug

I developed the [fold-preveiw](https://github.com/anuvyklack/fold-preview.nvim) plugin.

Today I updated to Neovim v0.8.2 and my
[fold-preveiw](https://github.com/anuvyklack/fold-preview.nvim)
plugin stopped working as it should. I haven't updated the plugin since November.

The plugin should open float window with folded region preview indented the same
way the folded code is. But now all float windows are opened in the zero column.

[I am not the only one who faced this issue.](https://github.com/anuvyklack/fold-preview.nvim/issues/4)

The plugin is quite simple and containes only one file.
[ The code part that open a window.](https://github.com/anuvyklack/fold-preview.nvim/blob/dcc6de437d3d049d46cc7c6a5d46f5729858441f/lua/fold-preview.lua#L136-L151)

I made some research and found that the `nvim_open_win` api function is broken.

I open Neovim with `nvim --clean` and execute next command in different places of the buffer
```lua
:lua vim.api.nvim_open_win(0, false, { 
    anchor = 'NW',
    border = 'rounded',
    relative = 'win',
    bufpos = vim.api.nvim_win_get_cursor(0),
    width = 25,
    height = 10
})
```
It should open the float window under the cursor.

If there is a text in the place when float window should appear, then its position would be correct:

![Screenshot from 2023-01-11 17-36-45](https://user-images.githubusercontent.com/13056013/211859259-da9ff7cc-58e5-4efe-ad16-632d000d7799.png)
![Screenshot from 2023-01-11 17-39-57](https://user-images.githubusercontent.com/13056013/211859313-b23131cf-82c7-458d-b5a0-fc3d8e38e2c1.png)

But if there is no text, that float window sticks to the last letter
Look at the cursor position and at the float window position.

![Screenshot from 2023-01-11 17-41-04](https://user-images.githubusercontent.com/13056013/211859805-dec9de72-173f-45ef-b942-f585d659900a.png)
![Screenshot from 2023-01-11 17-41-48](https://user-images.githubusercontent.com/13056013/211859860-6d3ed470-33c6-484b-a90f-7d611a5fc298.png)

### Steps to reproduce

```
nvim --clean file.txt
```

Execute next command at the end of the most long line, so that all below lines are shorter,
```
:lua vim.api.nvim_open_win(0, false, { anchor = 'NW', border = 'rounded', relative = 'win', bufpos = vim.api.nvim_win_get_cursor(0), width = 25, height = 10 })
```

### Expected behavior

The floating window should open such that it left upper corner should appear one line under the cursor position.

### Neovim version (nvim -v)

0.8.2

### Vim (not Nvim) behaves the same?

-

### Operating system/version

fedora 36

### Terminal name/version

kitty 0.26.5

### $TERM environment variable

tmux-256color

### Installation

dnf

**Comment by justinmk** [comment]
> Today I updated to Neovim v0.8.2

is this a regression compared to 0.8.1, i.e. did it work for you in 0.8.1 (else please say which version worked for you)?



> If there is a text in the place when float window should appear, then its position would be correct:

so it looks like the issue depends on presence of text in the buffer.

"nvim_open_win is broken!" is a bit much, in the future please choose a more precise issue title.

**Comment by anuvyklack** [comment]
No. The problem is not the line ([and I subtracted one](https://github.com/anuvyklack/fold-preview.nvim/blob/dcc6de437d3d049d46cc7c6a5d46f5729858441f/lua/fold-preview.lua#L140)).

The problem is column position.

**Comment by anuvyklack** [comment]
@zeertzjq, you get it wrong, the issue is steel here!

**Comment by zeertzjq** [comment]
If the anchored buffer text position is in a fold than the previous behavior makes even less sense.

**Comment by anuvyklack** [comment]
Ok, maybe you're right, and there was a bug that I relied on.

**Comment by anuvyklack** [comment]
Anchored buffer text position was not inside fold, but first line of the fold, which is available, as I understand.
